DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs

```diff
@@ -188,5 +188,42 @@ public void UnlockAccount_unlocks_account_and_returns_to_page()
             A.CallTo(() => userDataService.UpdateAdminUserFailedLoginCount(1, 0)).MustHaveHappened();
             result.Should().BeRedirectToActionResult().WithActionName("Index");
         }
+
+        [Test]
+        public void DeactivateAdminUser_does_not_deactivate_admin_user_without_confirmation()
+        {
+            // Given
+            var adminUser = UserTestHelper.GetDefaultAdminUser(active: true);
+            A.CallTo(() => userDataService.GetAdminUserById(A<int>._)).Returns(UserTestHelper.GetDefaultAdminUser());
+            var deactivateViewModel = new DeactivateAdminViewModel(adminUser) { Confirm = false };
+            const string expectedErrorMessage = "You must confirm before deactivating this account.";
+
+            // When
+            var result = administratorController.DeactivateAdmin(1, deactivateViewModel);
+
+            // Then
+            result.Should().BeViewResult().WithDefaultViewName().ModelAs<DeactivateAdminViewModel>();
+            administratorController.ModelState[nameof(DeactivateAdminViewModel.Confirm)].Errors[0].ErrorMessage.Should()
+                .BeEquivalentTo(expectedErrorMessage);
+            A.CallTo(() => userDataService.DeactivateAdmin(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void DeactivateAdminUser_deactivate_admin_user_with_confirmation()
+        {
+            // Given
+            var adminUser = UserTestHelper.GetDefaultAdminUser(active: true);
+            A.CallTo(() => userDataService.GetAdminUserById(A<int>._)).Returns(UserTestHelper.GetDefaultAdminUser());
+            var deactivateViewModel = new DeactivateAdminViewModel(adminUser) { Confirm = true };
+
+            // When
+            var result = administratorController.DeactivateAdmin(1, deactivateViewModel);
+            var deactivatedAdminUser = result.Should().BeViewResult().Model as AdminUser;
+
+            // Then
+            A.CallTo(() => userDataService.DeactivateAdmin(A<int>._)).MustHaveHappened();

```

Comment:
We can safely be more specific here and replace `A<int>._` with `1` just to test that the data service method is called with the ID that is passed in (1), rather than any ID which would pass A<int>._.

Reply:
I have updated with arguments 1 as suggested